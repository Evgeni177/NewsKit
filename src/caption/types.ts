import {MQ} from '../utils/style';

export interface CaptionProps {
  captionText?: string;
  creditText?: string;
  overrides?: {
    typographyPreset?: MQ<string>;
    stylePreset?: MQ<string>;
    spaceStack?: MQ<string>;
    spaceInset?: MQ<string>;
    credit?: {
      typographyPreset?: MQ<string>;
      stylePreset?: MQ<string>;
    };
  };
}
