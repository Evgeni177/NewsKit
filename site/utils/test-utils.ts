import {
  renderToFragmentWithThemeFactory,
  renderWithThemeFactory,
} from 'newskit/test/test-utils';
import {docsThemeLight} from '../theme/doc-theme';

export const renderWithTheme = renderWithThemeFactory(docsThemeLight);

export const renderToFragmentWithTheme = renderToFragmentWithThemeFactory(
  docsThemeLight,
);
