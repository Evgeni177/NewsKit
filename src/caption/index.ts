export * from './caption';
export * from './caption-inset';
export * from './types';
