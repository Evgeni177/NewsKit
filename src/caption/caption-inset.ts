import {withDefaultProps} from '../utils/with-default-props';
import {Caption} from './caption';

export const CaptionInset = withDefaultProps(Caption, {}, 'captionInset');
