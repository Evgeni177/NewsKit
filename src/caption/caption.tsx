import React from 'react';
import {CaptionProps} from './types';
import {Block} from '../block';
import {getToken} from '../utils/get-token';
import {useTheme} from '../theme';
import {TextBlock} from '../text-block';

export const Caption: React.FC<CaptionProps> = ({
  overrides,
  children,
  creditText,
}) => {
  const theme = useTheme();
  const captionSpaceStack =
    creditText && getToken({theme, overrides}, 'caption', '', 'spaceStack');

  const captionStylePreset = getToken(
    {theme, overrides},
    'caption',
    '',
    'stylePreset',
  );

  const captionTypographyPreset = getToken(
    {theme, overrides},
    'caption',
    '',
    'typographyPreset',
  );

  const creditTypographyPreset = getToken(
    {theme, overrides},
    'caption.credit',
    'credit',
    'typographyPreset',
  );
  const creditStylePreset = getToken(
    {theme, overrides},
    'caption.credit',
    'credit',
    'stylePreset',
  );

  const captionInset = getToken(
    {theme, overrides},
    'caption',
    '',
    'spaceInset',
  );
  return (
    <Block overrides={{spaceInset: captionInset}}>
      <Block overrides={{spaceStack: captionSpaceStack}}>
        <TextBlock
          overrides={{
            stylePreset: captionStylePreset,
            typographyPreset: captionTypographyPreset,
          }}
        >
          {children}
        </TextBlock>
      </Block>
      {creditText && (
        <TextBlock
          overrides={{
            stylePreset: creditStylePreset,
            typographyPreset: creditTypographyPreset,
          }}
        >
          {creditText}
        </TextBlock>
      )}
    </Block>
  );
};

Caption.displayName = 'Caption';
