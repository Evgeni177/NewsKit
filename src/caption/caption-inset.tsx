import React from 'react';
import {withDefaultProps} from '../utils/with-default-props';
import {Caption} from './caption';
import {styled} from '../utils/style';
import {CaptionProps} from './types';

export const CaptionInset = withDefaultProps(Caption, {}, 'captionInset');

export const StyledDiv = styled.div`
  border: 1px solid black;
`;

export const CaptionInsetWithBorder: React.FC<CaptionProps> = props => (
  <StyledDiv>
    <CaptionInset {...props} />
  </StyledDiv>
);
