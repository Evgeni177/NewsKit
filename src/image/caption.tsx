import React from 'react';
import {Caption} from '../caption';
import {useTheme} from '../theme';
import {getToken} from '../utils/get-token';
import {ImageProps} from './types';

export const ImageCaption = ({
  captionText,
  creditText,
  overrides,
}: {
  captionText: string;
  creditText?: string;
  overrides?: ImageProps['overrides'];
}) => {
  const theme = useTheme();
  const captionSpaceStack =
    creditText &&
    getToken({theme, overrides}, 'image', 'caption', 'spaceStack');

  const captionStylePreset = getToken(
    {theme, overrides},
    'image',
    'caption',
    'stylePreset',
  );

  const captionTypographyPreset = getToken(
    {theme, overrides},
    'image',
    'caption',
    'typographyPreset',
  );

  const creditStylePreset = getToken(
    {theme, overrides},
    'image.caption.credit',
    'caption.credit',
    'stylePreset',
  );

  const creditTypographyPreset = getToken(
    {theme, overrides},
    'image.caption.credit',
    'caption.credit',
    'typographyPreset',
  );

  const captionSpaceInset = getToken(
    {theme, overrides},
    'image',
    'caption',
    'spaceInset',
  );

  const captionOverrides = {
    stylePreset: captionStylePreset,
    typographyPreset: captionTypographyPreset,
    spaceInset: captionSpaceInset,
    spaceStack: captionSpaceStack,
    credit: {
      stylePreset: creditStylePreset,
      typographyPreset: creditTypographyPreset,
    },
  };

  return (
    <Caption creditText={creditText} overrides={captionOverrides}>
      {captionText}
    </Caption>
  );
};
