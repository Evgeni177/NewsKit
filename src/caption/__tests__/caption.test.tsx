import {Caption} from '../caption';
import {CaptionInset, CaptionInsetWithBorder} from '../caption-inset';
import {renderToFragmentWithTheme} from '../../test/test-utils';

describe('Caption', () => {
  test('renders correctly with default props and caption and credit text', () => {
    const fragment = renderToFragmentWithTheme(Caption, {
      children: 'test',
      creditText: 'this is credit text',
    });
    expect(fragment).toMatchSnapshot();
  });
});

test('renders correctly with default props and only caption text', () => {
  const fragment = renderToFragmentWithTheme(Caption, {
    children: 'test',
  });

  expect(fragment).toMatchSnapshot();
});

test('renders correctly with overriden props and caption and credit text', () => {
  const fragment = renderToFragmentWithTheme(Caption, {
    children: 'test',
    creditText: 'this is credit text',
    overrides: {
      typographyPreset: 'utilitySubheading050',
      stylePreset: 'inkContrast',
      spaceStack: 'space090',
      credit: {
        typographyPreset: 'utilityMeta010',
        stylePreset: 'uppercaseInkContrast',
      },
    },
  });
  expect(fragment).toMatchSnapshot();
});

describe('CaptionInset', () => {
  test('renders correctly with default props and caption and credit text', () => {
    const fragment = renderToFragmentWithTheme(CaptionInset, {
      children: 'test',
      creditText: 'this is credit text',
    });
    expect(fragment).toMatchSnapshot();
  });

  test('renders correctly with border', () => {
    const fragment = renderToFragmentWithTheme(CaptionInsetWithBorder, {
      children: 'test',
      creditText: 'this is credit text',
    });
    expect(fragment).toMatchSnapshot();
  });

  test('renders correctly with default props and only caption text', () => {
    const fragment = renderToFragmentWithTheme(CaptionInset, {
      children: 'test',
    });
    expect(fragment).toMatchSnapshot();
  });

  test('renders correctly with overriden props and caption and credit text', () => {
    const fragment = renderToFragmentWithTheme(CaptionInset, {
      children: 'test',
      creditText: 'this is credit text',
      overrides: {
        typographyPreset: 'utilitySubheading050',
        stylePreset: 'inkContrast',
        spaceStack: 'space090',
        credit: {
          typographyPreset: 'utilityMeta010',
          stylePreset: 'uppercaseInkContrast',
        },
      },
    });
    expect(fragment).toMatchSnapshot();
  });
});
