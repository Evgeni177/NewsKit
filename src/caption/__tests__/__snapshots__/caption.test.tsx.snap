// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Caption renders correctly with default props and caption and credit text 1`] = `
<DocumentFragment>
  .emotion-1 {
  margin-bottom: 16px;
}

.emotion-0 {
  margin: 0;
  padding: 1px 0;
  color: #2e2e2e;
  font-family: "Noto Sans",sans-serif;
  font-size: 14px;
  line-height: 1.4285714285714286;
  font-weight: 400;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
}

.emotion-0 svg {
  fill: #2e2e2e;
}

.emotion-0::before {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-bottom: calc(-0.3642857142857143em + 1.2px);
}

.emotion-0::after {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-top: -0.4267857142857143em;
}

.emotion-2 {
  margin: 0;
  padding: 1px 0;
  color: #535353;
  text-transform: uppercase;
  font-family: "Noto Sans",sans-serif;
  font-size: 12px;
  line-height: 1.6666666666666667;
  font-weight: 500;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
}

.emotion-2 svg {
  fill: #535353;
}

.emotion-2::before {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-bottom: calc(-0.4833333333333334em + 1px);
}

.emotion-2::after {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-top: calc(-0.5458333333333334em + -0.5px);
}

<div
    class="emotion-3"
  >
    <div
      class="emotion-1"
    >
      <p
        class="emotion-0"
      >
        test
      </p>
    </div>
    <p
      class="emotion-2"
    >
      this is credit text
    </p>
  </div>
</DocumentFragment>
`;

exports[`CaptionInset renders correctly with default props and caption and credit text 1`] = `
<DocumentFragment>
  .emotion-1 {
  margin-bottom: 16px;
}

.emotion-0 {
  margin: 0;
  padding: 1px 0;
  color: #2e2e2e;
  font-family: "Noto Sans",sans-serif;
  font-size: 14px;
  line-height: 1.4285714285714286;
  font-weight: 400;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
}

.emotion-0 svg {
  fill: #2e2e2e;
}

.emotion-0::before {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-bottom: calc(-0.3642857142857143em + 1.2px);
}

.emotion-0::after {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-top: -0.4267857142857143em;
}

.emotion-2 {
  margin: 0;
  padding: 1px 0;
  color: #535353;
  text-transform: uppercase;
  font-family: "Noto Sans",sans-serif;
  font-size: 12px;
  line-height: 1.6666666666666667;
  font-weight: 500;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
}

.emotion-2 svg {
  fill: #535353;
}

.emotion-2::before {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-bottom: calc(-0.4833333333333334em + 1px);
}

.emotion-2::after {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-top: calc(-0.5458333333333334em + -0.5px);
}

@media screen and (max-width:767px) {
  .emotion-3 {
    padding: 16px;
  }
}

@media screen and (min-width:768px) {
  .emotion-3 {
    padding: 24px;
  }
}

<div
    class="emotion-3"
  >
    <div
      class="emotion-1"
    >
      <p
        class="emotion-0"
      >
        test
      </p>
    </div>
    <p
      class="emotion-2"
    >
      this is credit text
    </p>
  </div>
</DocumentFragment>
`;

exports[`CaptionInset renders correctly with default props and only caption text 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  padding: 1px 0;
  color: #2e2e2e;
  font-family: "Noto Sans",sans-serif;
  font-size: 14px;
  line-height: 1.4285714285714286;
  font-weight: 400;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
}

.emotion-0 svg {
  fill: #2e2e2e;
}

.emotion-0::before {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-bottom: calc(-0.3642857142857143em + 1.2px);
}

.emotion-0::after {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-top: -0.4267857142857143em;
}

@media screen and (max-width:767px) {
  .emotion-2 {
    padding: 16px;
  }
}

@media screen and (min-width:768px) {
  .emotion-2 {
    padding: 24px;
  }
}

<div
    class="emotion-2"
  >
    <div
      class="emotion-1"
    >
      <p
        class="emotion-0"
      >
        test
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`CaptionInset renders correctly with overriden props and caption and credit text 1`] = `
<DocumentFragment>
  .emotion-1 {
  margin-bottom: 64px;
}

.emotion-0 {
  margin: 0;
  padding: 1px 0;
  color: #0a0a0a;
  font-family: "Noto Sans",sans-serif;
  font-size: 24px;
  line-height: 1.1666666666666667;
  font-weight: 400;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
}

.emotion-0 svg {
  fill: #0a0a0a;
}

.emotion-0::before {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-bottom: calc(-0.2333333333333334em + 1px);
}

.emotion-0::after {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-top: calc(-0.2958333333333334em + 1px);
}

.emotion-2 {
  margin: 0;
  padding: 1px 0;
  color: #0a0a0a;
  text-transform: uppercase;
  font-family: "Noto Sans",sans-serif;
  font-size: 12px;
  line-height: 1.6666666666666667;
  font-weight: 500;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
}

.emotion-2 svg {
  fill: #0a0a0a;
}

.emotion-2::before {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-bottom: calc(-0.4833333333333334em + 1px);
}

.emotion-2::after {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-top: calc(-0.5458333333333334em + -0.5px);
}

@media screen and (max-width:767px) {
  .emotion-3 {
    padding: 16px;
  }
}

@media screen and (min-width:768px) {
  .emotion-3 {
    padding: 24px;
  }
}

<div
    class="emotion-3"
  >
    <div
      class="emotion-1"
    >
      <p
        class="emotion-0"
      >
        test
      </p>
    </div>
    <p
      class="emotion-2"
    >
      this is credit text
    </p>
  </div>
</DocumentFragment>
`;

exports[`renders correctly with default props and only caption text 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  padding: 1px 0;
  color: #2e2e2e;
  font-family: "Noto Sans",sans-serif;
  font-size: 14px;
  line-height: 1.4285714285714286;
  font-weight: 400;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
}

.emotion-0 svg {
  fill: #2e2e2e;
}

.emotion-0::before {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-bottom: calc(-0.3642857142857143em + 1.2px);
}

.emotion-0::after {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-top: -0.4267857142857143em;
}

<div
    class="emotion-1"
  >
    <div
      class="emotion-1"
    >
      <p
        class="emotion-0"
      >
        test
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders correctly with overriden props and caption and credit text 1`] = `
<DocumentFragment>
  .emotion-1 {
  margin-bottom: 64px;
}

.emotion-0 {
  margin: 0;
  padding: 1px 0;
  color: #0a0a0a;
  font-family: "Noto Sans",sans-serif;
  font-size: 24px;
  line-height: 1.1666666666666667;
  font-weight: 400;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
}

.emotion-0 svg {
  fill: #0a0a0a;
}

.emotion-0::before {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-bottom: calc(-0.2333333333333334em + 1px);
}

.emotion-0::after {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-top: calc(-0.2958333333333334em + 1px);
}

.emotion-2 {
  margin: 0;
  padding: 1px 0;
  color: #0a0a0a;
  text-transform: uppercase;
  font-family: "Noto Sans",sans-serif;
  font-size: 12px;
  line-height: 1.6666666666666667;
  font-weight: 500;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
}

.emotion-2 svg {
  fill: #0a0a0a;
}

.emotion-2::before {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-bottom: calc(-0.4833333333333334em + 1px);
}

.emotion-2::after {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-top: calc(-0.5458333333333334em + -0.5px);
}

<div
    class="emotion-3"
  >
    <div
      class="emotion-1"
    >
      <p
        class="emotion-0"
      >
        test
      </p>
    </div>
    <p
      class="emotion-2"
    >
      this is credit text
    </p>
  </div>
</DocumentFragment>
`;
